src/main/java/co/rsk/federate/signing/SignerVersion.java

```diff
@@ -0,0 +1,20 @@
+package co.rsk.federate.signing;
+
+public enum SignerVersion {
```

Comment:
What about making the enum implement the `Comparable<SignerVersion>` interface. This way we can do conditional checks. 

```java
    @Override
    public int compareTo(SignerVersion other) {
        return Integer.compare(this.version, other.version);
    }
```

Reply:
This is not possible since the `enum` class already implements comparable and sets the implemented method as final. This means that the method cannot be overridden.